README.md

```diff
@@ -56,186 +50,416 @@ npm install @dojo/has
+### Basic Widgets
+
+Dojo 2 applications use a concept called Virtual DOM (vdom) to represent what should be shown on the screen. These vdom nodes are plain javascript objects that are not expensive to create, unlike browser DOM elements. Dojo 2 uses these vdom elements to synchronize and update the browser DOM so that application shows the expected view.
+
+There are two types of vdom within Dojo 2, the first are pure representations of DOM elements and are the fundamental building blocks of all Dojo 2 applications. These are called `HNode`s and are created using the `v()` function available from the `@dojo/widget-core/d` module.
+
+The following will create a `HNode` that represents a simple `div` DOM element, with a text node child: `Hello, Dojo 2!`:
 
 ```ts
-class MyWidget extends WidgetBase<WidgetProperties> {
-	render() {
-       return v('h1', [ 'Hello, Dojo!' ]);
+v('div', [ 'Hello, Dojo 2!' ])
+```
+
+The second vdom type, `WNode`s represent widgets. A widget is a class that extends `WidgetBase` from `@dojo/widget-core/WidgetBase` and implements a `render` function that returns one of the Dojo 2 vdom types (known as a `DNode`). Widgets are used to represent reusable, independent sections of a Dojo 2 application.
+
+The following returns the `HNode` example from above from the `render` function:
+
+```ts
+class HelloDojo extends WidgetBase {
+	protected render(): DNode {
+		return v('div', [ 'Hello, Dojo 2!' ]);
 	}
 }
+```
+
+#### Rendering a Widget in the DOM
+
+To display your new component in the browser you will need to decorate it with some functionality needed to "project" the widget into the browser. This is done using the `ProjectorMixin` from `@dojo/widget-core/mixins/Projector`.
+
+```ts
+const Projector = ProjectorMixin(HelloDojo);
+const projector = new Projector();
 
-const Projector = ProjectorMixin(MyWidget);
+projector.append();
+```
+
+By default the projector will attach the widget to the `document.body` in the DOM, but this can be overridden by passing a reference to the Element to attach to.
+
+Consider the following in your HTML file:
+
+```html
+<div id="my-app"></div>
+```
+
+You can target this Element like so:
+
+```ts
+const root = document.getElementById('my-app');
+const Projector = ProjectorMixin(HelloDojo);
 const projector = new Projector();
 
 projector.append(root);
 ```
 
-This code renders a `h1` element onto the page, that says "Hello, Dojo!".
+#### Widgets and Properties
+
+We have created a widget used to project our `HNode`s into the DOM, however widgets can be composed of other widgets and `properties` are used to determine if a widget needs to be re-rendered.
 
-### Overview
+Properties are available on the the widget instance, defined by an interface and passed as a generic to the `WidgetBase` class when creating your custom component. The properties interface should extend the base `WidgetProperties` provided from `@dojo/widget-core/interfaces`:
```

Comment:
generic -> `generic` (and perhaps worth a link to https://www.typescriptlang.org/docs/handbook/generics.html ?)